contracts/protocol/facets/SellerHandlerFacet.sol

```diff
@@ -94,73 +94,178 @@ contract SellerHandlerFacet is SellerBase {
         address sender = msgSender();
 
         // Check that caller is authorized to call this function
-        if (seller.admin == address(0)) {
+        if (authToken.tokenType != AuthTokenType.None) {
             address authTokenContract = lookups.authTokenContracts[authToken.tokenType];
             address tokenIdOwner = IERC721(authTokenContract).ownerOf(authToken.tokenId);
             require(tokenIdOwner == sender, NOT_ADMIN);
         } else {
             require(seller.admin == sender, NOT_ADMIN);
         }
 
-        // Check that the passed in addresses are unique to one seller id across all roles -- not used or are used by this seller id.
-        // Checking this seller id is necessary because one or more addresses may not change
-        {
-            uint256 check1 = lookups.sellerIdByOperator[_seller.operator];
-            uint256 check2 = lookups.sellerIdByOperator[_seller.clerk];
-            uint256 check3 = lookups.sellerIdByAdmin[_seller.operator];
-            uint256 check4 = lookups.sellerIdByAdmin[_seller.clerk];
-            uint256 check5 = lookups.sellerIdByClerk[_seller.operator];
-            uint256 check6 = lookups.sellerIdByClerk[_seller.clerk];
-            require(
-                (check1 == 0 || check1 == _seller.id) &&
-                    (check2 == 0 || check2 == _seller.id) &&
-                    (check3 == 0 || check3 == _seller.id) &&
-                    (check4 == 0 || check4 == _seller.id) &&
-                    (check5 == 0 || check5 == _seller.id) &&
-                    (check6 == 0 || check6 == _seller.id),
-                SELLER_ADDRESS_MUST_BE_UNIQUE
-            );
-        }
+        sellerPreUpdateChecks(_seller.id, _seller.admin, lookups);
+        sellerPreUpdateChecks(_seller.id, _seller.operator, lookups);
+        sellerPreUpdateChecks(_seller.id, _seller.clerk, lookups);
+
+        // Clean old seller pending update data if exists
+        delete lookups.sellerPendingUpdates[_seller.id];
+
+        bool needsApproval;
 
         // Admin address or AuthToken data must be present in parameters. A seller can have one or the other. Check passed in parameters
-        if (_seller.admin == address(0)) {
+        if (_authToken.tokenType != AuthTokenType.None) {
             // Check that auth token is unique to this seller
             uint256 check = lookups.sellerIdByAuthToken[_authToken.tokenType][_authToken.tokenId];
             require(check == 0 || check == _seller.id, AUTH_TOKEN_MUST_BE_UNIQUE);
+
+            delete lookups.sellerIdByAuthToken[authToken.tokenType][authToken.tokenId];
+
+            // Store auth token
+            authToken.tokenId = _authToken.tokenId;
+            authToken.tokenType = _authToken.tokenType;
+
+            // Store seller by auth token reference
+            lookups.sellerIdByAuthToken[_authToken.tokenType][_authToken.tokenId] = _seller.id;
```

Comment:
When we first implemented `createSeller`, we did not check that any addresses == caller or that the caller owns the AuthToken. We therefore also didn't check in `updateSeller` that  the new passed in AuthToken was owned by the caller. We only checked that the caller was owner of the STORED AuthToken to validate if the caller can call the function.  Now that we do check that the caller owns the AuthToken in `createSeller`, it seems to me we should also check that the caller owns the new passed in AuthToken in `updateSeller`, especially if there is no approval step for changing the AuthToken.

@zajck @anajuliabit @cliffhall Do you agree?

Reply:
Well that would mean that `caller = ownerOf(storedAuthToken)` and `caller = ownerOf(newAuthToken)` simultaneously, which forces `ownerOf(storedAuthToken) = ownerOf(newAuthToken)` 

But what if you are trying to change AuthToken because you actually want to transfer admin rights to some other address?
Maybe we need also approval step for changing the auth token :thinking:

Reply:
@zajck good point, I agree. I think we also need approval for changing the auth token. What do you think @cliffhall @mischat @hswopeams ?

Reply:
Yes, I agree. The owner of the new auth token should confirm their opt-in, just as if an address had been supplied.